transformer/classes/dockerfile/pythondockerfilegenerator.go

```diff
@@ -0,0 +1,248 @@
+	}
+	return pythonMainFiles, nil
+}
+
+// findDjangoDependency checks for django dependency in the requirements.txt file
+func findDjangoDependency(reqTxtFilePath string) bool {
+	reqTxtFile, err := os.Open(reqTxtFilePath)
+	if err != nil {
+		logrus.Debugf("Failed to open the file %s", reqTxtFilePath)
+	}
+	scanner := bufio.NewScanner(reqTxtFile)
+	scanner.Split(bufio.ScanLines)
+	for scanner.Scan() {
+		if djangoDependencyRegex.MatchString(scanner.Text()) {
+			return true
+		}
+	}
+	return false
+}
+
+// getMainPythonFileForService returns the main file used by a service
+func getMainPythonFileForService(mainPythonFilesPath []string, baseDir string, serviceName string) string {
+	var mainPythonFilesRelPath []string
+	for _, mainPythonFilePath := range mainPythonFilesPath {
+		if mainPythonFileRelPath, err := filepath.Rel(baseDir, mainPythonFilePath); err == nil {
+			mainPythonFilesRelPath = append(mainPythonFilesRelPath, mainPythonFileRelPath)
+		}
+	}
+	return qaengine.FetchSelectAnswer(common.ConfigServicesKey+common.Delim+serviceName+common.Delim+common.ConfigMainPythonFileForServiceKeySegment, fmt.Sprintf("Select the main file to be used for the service %s :", serviceName), []string{fmt.Sprintf("Selected main file will be used for the service %s", serviceName)}, mainPythonFilesRelPath[0], mainPythonFilesRelPath)
+}
+
+// DirectoryDetect runs detect in each sub directory
+func (t *PythonDockerfileGenerator) DirectoryDetect(dir string) (services map[string][]transformertypes.Artifact, err error) {
+
+	pythonFiles, err := common.GetFilesByExtInCurrDir(dir, []string{pythonExt})
+	if err != nil {
+		logrus.Errorf("Error while finding python files %s", err)
+		return nil, nil
+	}
+	if len(pythonFiles) == 0 {
+		return nil, nil
+	}
+	mainPythonFilesPath, err := findMainScripts(pythonFiles)
+	if err != nil {
+		return nil, err
+	}
+	if len(mainPythonFilesPath) != 0 {
+		var serviceName, requirementsTxtPath string
+		var isDjangoProject bool
+		var detectedPorts []int32
+		detectedPorts = append(detectedPorts, 8080) //TODO: Write parser to parse and identify port
+		requirementsTxtFiles, err := common.GetFilesByName(dir, []string{requirementsTxtFile}, nil)
+		if err != nil {
+			logrus.Debugf("Cannot get the requirements.txt file: %s", err)
+		}
+		if len(requirementsTxtFiles) == 1 {
+			requirementsTxt, err := filepath.Rel(dir, requirementsTxtFiles[0])
+			if err != nil {
+				logrus.Errorf("Error in getting the relative path %s", err)
+			} else {
+				requirementsTxtPath = requirementsTxt
+				isDjangoProject = findDjangoDependency(requirementsTxtFiles[0])
+			}
+		}
+		services = map[string][]transformertypes.Artifact{
+			serviceName: {{
+				Paths: map[string][]string{
+					artifacts.ProjectPathPathType: {dir},
+					MainPythonFilesPathType:       mainPythonFilesPath,
+				}, Configs: map[string]interface{}{
+					DjangoProjectConfigType:   isDjangoProject,
+					RequirementsTxtConfigType: requirementsTxtPath,
```

Comment:
Paths should go in Paths and not in Configs.

Reply:
Done.